colossalai/graph/passes/shard_1d_pass.py

```diff
@@ -0,0 +1,98 @@
+import torch
+from torch.fx.node import map_arg
+from torch.fx.node import Node
+from torch.fx.passes.split_module import split_module
+
+import colossalai
+from colossalai.context import ParallelMode
+from colossalai.core import global_context as gpc
+
+
+def all_gather_function(input_):
+    world_size = gpc.get_world_size(ParallelMode.PARALLEL_1D)
+    rank = gpc.get_local_rank(ParallelMode.PARALLEL_1D)
+    tensor_list = [torch.empty_like(input_) for _ in range(world_size)]
+    tensor_list[rank] = input_
+    group = gpc.get_group(ParallelMode.PARALLEL_1D)
+    torch.distributed.all_gather(tensor_list, input_, group=group)
+    output = torch.cat(tensor_list, dim=-1).contiguous()
+    return output
+
+
+def all_reduce_function(input_):
+    if gpc.get_world_size(ParallelMode.PARALLEL_1D) == 1:
+        return input_
+    torch.distributed.all_reduce(input_, group=gpc.get_group(ParallelMode.PARALLEL_1D))
+    return input_
+
+
+def weight_split(weight, dim):
+    num_partition = gpc.get_world_size(ParallelMode.TENSOR)
+    shape = weight.shape
+    length = shape[dim] // num_partition
+    sharded_weight_list = []
+    for i in range(num_partition):
+        sharded_weight_list.append(weight.narrow(dim, i * length, length))
+    return sharded_weight_list[gpc.get_local_rank(ParallelMode.PARALLEL_1D)]
```

Comment:
I would suggest to utilize the `dist_spec` in ColoTensor.

Reply:
If `ColoTensor` is not used for now, let's add a TODO comment for future replacement of torch tensor with colotensor.